File: Tests/DeepChildrenTests.m
//
//  DeepChildrenTests.m
//  RaptureXML
//
//  Created by John Blanco on 9/24/11.
//  Copyright (c) 2011 Rapture In Venice. All rights reserved.
//

#import "RaptureTestSuite.h"

@interface DeepChildrenTests : RaptureTestSuite {
}

@end



@implementation DeepChildrenTests

- (void)testQuery {
    RXMLElement *rxml = [self testXML];
    __block NSInteger i = 0;
    
    // count the players
    RXMLElement *players = [rxml child:@"players"];
    NSArray *children = [players children:@"player"];
    
    [rxml iterateElements:children usingBlock: ^(RXMLElement *e) {
        i++;
    }];    
    
    XCTAssertEqual(i, 9);
}

- (void)testDeepChildQuery {
    RXMLElement *rxml = [self testXML];

    // count the players
    RXMLElement *coachingYears = [rxml child:@"players.coach.experience.years"];
    
    XCTAssertEqual(coachingYears.textAsInt, 1);
}

- (void)testDeepChildQueryWithWildcard {
    RXMLElement *rxml = [self testXML];
    
    // count the players
    RXMLElement *coachingYears = [rxml child:@"players.coach.experience.teams.*"];
    
    // first team returned
    XCTAssertEqual(coachingYears.textAsInt, 53);
}

@end

File: Swift/RaptureXMLTests/DeepChildrenTests.swift
//
//  DeepChildrenTests.swift
//  RaptureXML-Swift
//
//  Created by Brett Lamy on 2/12/15.
//  Copyright (c) 2015 Brett Lamy. All rights reserved.
//


import UIKit
import XCTest

class DeepChildrenTests: XCTestCase {
    
    func testInterruptedText() {
        let rxml = RXMLElement(fromXMLFile:"players.xml");
        var i = 0;
        
        let players = rxml.child("players");
        let children : NSArray = players.children("player");
        
        rxml.iterateElements(children, usingBlock: { (e : RXMLElement!) -> Void in
            i += 1;
        });
        
        XCTAssertEqual(i, 9);
    }
    
    func testDeepChildQuery() {
        let rxml = RXMLElement(fromXMLFile:"players.xml");
        
        let coachingYears = rxml.child("players.coach.experience.years");
        
        XCTAssertEqual(coachingYears.textAsInt, 1);
    }
    
    func testDeepChildQueryWithWildcard() {
        let rxml = RXMLElement(fromXMLFile:"players.xml");
        
        let coachingYears = rxml.child("players.coach.experience.teams.*");
        
        XCTAssertEqual(coachingYears.textAsInt, 53);
    }

}
